subsys/bluetooth/audio/tbs_client.c

```diff
@@ -0,0 +1,2166 @@
+static uint8_t notify_handler(struct bt_conn *conn,
+			      struct bt_gatt_subscribe_params *params,
+			      const void *data, uint16_t length)
+{
+	uint16_t handle = params->value_handle;
+	struct bt_tbs_instance *tbs_inst = lookup_inst_by_handle(conn,
+								       handle);
+
+	if (data == NULL) {
+		BT_DBG("[UNSUBSCRIBED] 0x%04X", params->value_handle);
+		params->value_handle = 0U;
+		if (tbs_inst != NULL) {
+			tbs_inst->subscribe_cnt--;
+		}
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	if (tbs_inst != NULL) {
+		if (IS_ENABLED(CONFIG_BT_TBS_CLIENT_GTBS) && tbs_inst->gtbs) {
+			BT_DBG("GTBS");
+		} else {
+			BT_DBG("Index %u", tbs_inst->index);
+		}
+		BT_HEXDUMP_DBG(data, length, "notify handler value");
+
+		if (handle == tbs_inst->provider_name_handle) {
+			provider_name_notify_handler(conn, tbs_inst, data,
+						     length);
+		} else if (handle == tbs_inst->technology_handle) {
+			technology_notify_handler(conn, tbs_inst, data, length);
+		} else if (handle == tbs_inst->signal_strength_handle) {
+			signal_strength_notify_handler(conn, tbs_inst, data,
+						       length);
+		} else if (handle == tbs_inst->status_flags_handle) {
+			status_flags_notify_handler(conn, tbs_inst, data,
+						    length);
+		} else if (handle == tbs_inst->current_calls_handle) {
+			current_calls_notify_handler(conn, tbs_inst, data,
+						     length);
+		} else if (handle == tbs_inst->in_uri_handle) {
+			incoming_uri_notify_handler(conn, tbs_inst, data,
+						    length);
+		} else if (handle == tbs_inst->call_state_handle) {
+			call_state_notify_handler(conn, tbs_inst, data, length);
+		} else if (handle == tbs_inst->call_cp_handle) {
+			call_cp_notify_handler(conn, tbs_inst, data, length);
+		} else if (handle == tbs_inst->termination_reason_handle) {
+			termination_reason_notify_handler(conn, tbs_inst, data,
+							  length);
+		} else if (handle == tbs_inst->in_call_handle) {
+			in_call_notify_handler(conn, tbs_inst, data, length);
+		} else if (handle == tbs_inst->friendly_name_handle) {
+			friendly_name_notify_handler(conn, tbs_inst, data,
+						     length);
+		}
+	} else {
+		BT_DBG("Notification/Indication on unknown TBS inst");
+	}
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static int tbs_client_common_call_control(struct bt_conn *conn, uint8_t inst_index,
+				   uint8_t call_index, uint8_t opcode)
+{
+	const struct bt_tbs_instance *inst = get_inst_by_index(conn, inst_index);
+	struct bt_tbs_call_cp_acc common;
+
+	if (inst == NULL) {
+		return -EINVAL;
+	}
+
+	if (inst->call_cp_handle == 0) {
+		BT_DBG("Handle not set");
+		return -EINVAL;
+	}
+
+	common.opcode = opcode;
+	common.call_index = call_index;
+
+	return bt_gatt_write_without_response(conn, inst->call_cp_handle,
+					      &common, sizeof(common), false);
+}
+
+static uint8_t read_bearer_provider_name_cb(struct bt_conn *conn,
+						uint8_t err,
+						struct bt_gatt_read_params *params,
+						const void *data,
+						uint16_t length)
+{
```

Comment:
Misalignment with the split lines here as well (there are more places in this file, but I won't comment on them separately)